optuna/trial/_fixed.py

```diff
@@ -108,16 +108,23 @@ def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float)
         discrete = distributions.DiscreteUniformDistribution(low=low, high=high, q=q)
         return self._suggest(name, discrete)
 
-    def suggest_int(self, name, low, high, step=1, log=False):
-        # type: (str, int, int, int, bool) -> int
-        if log:
-            sample = self._suggest(
-                name, distributions.IntLogUniformDistribution(low=low, high=high, step=step)
-            )
+    def suggest_int(self, name: str, low: int, high: int, step: int = 1, log: bool = False) -> int:
+        if step != 1:
+            if log:
+                raise ValueError("The parameter `step >= 2` is not supported when `log` is True.")
+            else:
+                sample = self._suggest(
+                    name, distributions.IntUniformDistribution(low=low, high=high, step=step)
+                )
         else:
-            sample = self._suggest(
-                name, distributions.IntUniformDistribution(low=low, high=high, step=step)
-            )
+            if log:
+                sample = self._suggest(
+                    name, distributions.IntLogUniformDistribution(low=low, high=high, step=step)
+                )
+            else:
+                sample = self._suggest(
+                    name, distributions.IntUniformDistribution(low=low, high=high, step=step)
+                )
```

Comment:
A nit but you can DRY some code by assigning the distribution type to a local variable in your conditions and have a single call to `self._suggest` as all other arguments and the function calls are identical.